src/sage/topology/moment_angle_complex.py

```diff
@@ -814,3 +819,712 @@ def has_trivial_lowest_deg_massey_product(self):
+            sage: H.basis()
+            Finite family {(0, 0): h^{0,0}, (7, 0): h^{7,0}}
+            sage: H.basis(5)
+            Finite family {}
+            sage: H.basis(7)
+            Finite family {(7, 0): h^{7,0}}
+            sage: H.basis()[7, 0]
+            h^{7,0}
+            sage: H.basis(8)
+            Finite family {}
+        """
+        if d is None:
+            return Family(self._indices, self.monomial)
+        else:
+            indices = [(d, i) for i in self._graded_indices.get(d, [])]
+            return Family(indices, self.monomial)
+
+    def degree_on_basis(self, i):
+        """
+        Return the degree of the basis element indexed by ``i``.
+
+        EXAMPLES::
+
+            sage: Z = MomentAngleComplex([[1,2], [3,4,5], [1,5], [2,3]])
+            sage: H = Z.cohomology_ring(GF(5))
+            sage: H.degree_on_basis((4, 0))
+            4
+        """
+        return i[0]
+
+    def _repr_(self):
+        """
+        Return a string representation of ``self``.
+
+        TESTS::
+
+            sage: MomentAngleComplex([[1,2,3], [3,4]]).cohomology_ring()
+            Cohomology module of Moment-angle complex of Simplicial complex
+            with vertex set (1, 2, 3, 4) and facets {(3, 4), (1, 2, 3)}
+            over Rational Field
+        """
+        return "Cohomology module of {} over {}".format(self._complex, self.base_ring())
+
+    def _repr_term(self, i):
+        """
+        Return ``'h^{i[0],i[1]}'``, for the basis element indexed by ``i``.
+
+        EXAMPLES::
+
+            sage: Z = MomentAngleComplex([[1,2], [3,4,5], [1,5], [2,3]])
+            sage: Z = MomentAngleComplex([[1,2], [3,4,5], [1,4], [2,3,5]])
+            sage: H = Z.cohomology_ring(GF(7))
+            sage: H.basis()[3, 2]
+            h^{3,2}
+        """
+        return 'h^{{{},{}}}'.format(i[0], i[1])
+
+    _latex_term = _repr_term
+
+    def one(self):
+        """
+        Return the multiplicative identity element.
+
+        EXAMPLES::
+
+            sage: Z = MomentAngleComplex([[1,2], [3,4], [2,3,5], [1,5], [4,5]])
+            sage: H = Z.cohomology_ring()
+            sage: H.one()
+            h^{0,0}
+            sage: all(H.one() * x == x == x * H.one() for x in H.basis())
+            True
+        """
+        one = self._base_ring.one()
+        d = {(0, i): one for i in self._graded_indices[0]}
+        return self._from_dict(d, remove_zeros=False)
+
+    def complex(self):
+        """
+        Return the moment-angle complex associated with ``self``.
+
+        EXAMPLES::
+
+            sage: S2 = simplicial_complexes.Sphere(2)
+            sage: H = MomentAngleComplex(S2).cohomology_ring()
+            sage: H.complex()
+            Moment-angle complex of Simplicial complex with vertex set
+            (0, 1, 2, 3) and facets {(0, 1, 2), (0, 1, 3), (0, 2, 3), (1, 2, 3)}
+        """
+        return self._complex
+
+    @cached_method
+    def _to_cycle_on_basis(self, i):
```

Comment:
```suggestion
    def _to_cocycle_on_basis(self, i):
```
Since it is returning cocycles.